components/announcement_bar/overage_users_banner/index.tsx

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import {GlobalState} from 'types/store';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import AnnouncementBar from 'components/announcement_bar/default_announcement_bar';
+import {calculateOverageUserActivated} from 'utils/overage_team';
+import {isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
+import {PreferenceType} from '@mattermost/types/preferences';
+import {LicenseLinks, StatTypes, Preferences, AnnouncementBarTypes} from 'utils/constants';
+
+import './overage_users_banner.scss';
+
+type AdminHasWatchItArgs = {
+    preferenceName: string;
+    overagePreferences: PreferenceType[];
+    isWarningBanner: boolean;
+}
+
+const adminHasWatchIt = ({preferenceName, overagePreferences, isWarningBanner}: AdminHasWatchItArgs): boolean => {
+    if (isWarningBanner) {
+        return overagePreferences.find((value) => value.name === preferenceName) !== undefined;
+    }
+
+    return false;
+};
+
+const OverageUsersBanner = () => {
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+    const stats = useSelector((state: GlobalState) => state.entities.admin.analytics) || {};
+    const isAdmin = useSelector(isCurrentUserSystemAdmin);
+    const user = useSelector(getCurrentUser);
+    const license = useSelector(getLicense);
+    const seatsPurchased = parseInt(license.Users, 10);
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const getPreferencesCategory = makeGetCategory();
+    const currentUser = useSelector((state: GlobalState) => getCurrentUser(state));
+    const overagePreferences = useSelector((state: GlobalState) => getPreferencesCategory(state, Preferences.OVERAGE_USERS_BANNER));
+    const activeUsers = ((stats || {})[StatTypes.TOTAL_USERS]) as number || 0;
+    const {
+        isBetween5PercerntAnd10PercentPurchasedSeats,
+        isOver10PercerntPurchasedSeats,
+    } = calculateOverageUserActivated({
+        activeUsers,
+        seatsPurchased,
+    });
+    const prefixPreferences = isOver10PercerntPurchasedSeats ? 'error' : 'warn';
+    const preferenceName = `${prefixPreferences}_overage_seats_${seatsPurchased}`;
```

Comment:
Neat. If I understand, this will make it so that each time they add a new user, they will re-see the banner until they dismiss it again (and if they already dismissed it and go down to a number they have already seen, they won't see it again).

Reply:
Exactly, and in the second scenario only when they are over the 10% threshold then they will see it again.

Reply:
Each time they add a new user it will show up again? The preference seems to always be the same for the combination `warn` + `seatsPurchased`

Reply:
it won't show up again, the warning state. Only when they buy new seats and reach overage users again.

Reply:
Should we use the license id here instead to be more precise? There is a chance they purchase a new license with the same amount of seats